docs/contributing/coding-guidelines/ros-nodes/directory-structure.md

```diff
@@ -1,75 +1,221 @@
 # Directory structure
 
-!!! warning
+This document describes the directory structure of ROS nodes within Autoware.
 
-    Under Construction
+We'll use the package `autoware_gnss_poser` as an example.
 
 ## C++ package
 
+### Entire structure
+
+- This is a reference on how the entire package might be structured.
+- A package may not have all the directories shown here.
+
 ```txt
-<package_name>
+autoware_gnss_poser
+├─ package.xml
+├─ CMakeLists.txt
+├─ README.md
+│
 ├─ config
-│   ├─ foo_ros.param.yaml
-│   └─ foo_non_ros.yaml
+│   ├─ gnss_poser.param.yaml
+│   └─ another_non_ros_config.yaml
+│
+├─ schema
+│   └─ gnss_poser.schema.json
+│
 ├─ doc
 │   ├─ foo_document.md
 │   └─ foo_diagram.svg
-├─ include
-│   └─ <package_name>
-│       └─ foo_public.hpp
-├─ launch
-│   ├─ foo.launch.xml
-│   └─ foo.launch.py
-├─ schema
-│   └─ foo_node.schema.json
+│
+├─ include  # for exporting headers
+│   └─ autoware
+│       └─ gnss_poser
+│           └─ exported_header.hpp
+│
 ├─ src
-│   ├─ foo_node.cpp
-│   ├─ foo_node.hpp
-│   └─ foo_private.hpp
-├─ test
-│   └─ test_foo.cpp
+│   ├─ include
```

Comment:
I think `include` might be confusing, given that we already have a top level include. In other packages I've seen `detail` (e.g. https://github.com/ros2/rclcpp/tree/rolling/rclcpp/src/rclcpp)

Reply:
But in that detail folder doesn't have any **.hpp** files.

Also in the latest form of the documentation, we have left the organization of `src` folder up to developer. Do you think it's enough?

Here: https://github.com/autowarefoundation/autoware-documentation/blob/9ba71b95d580d47e711653315fd43c8616161241/docs/contributing/coding-guidelines/ros-nodes/directory-structure.md?plain=1#L170-L198